### PR TITLE
internal/dockerfiles: Follow dockerfile security best practices

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
Follows a few recommendations from a security code review

### Motivation
Align with docker file security best practices (these changes are non-exhaustive of all the security improvements, however they were the easiest / fastest to implement so I thought it prudent to just get these out there since they seem very low risk of breaking changes)

### Describe how to test/QA your changes
ensure the docker files still function as expected

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]
Do we have a guide for upgrading the Go version in dd-trace-go somewhere? It's worth documenting an extra step to bump the Go version in the dockerfiles.

**Review by eliottness — APPROVED** [review]

File: internal/apps/unit-of-work/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-FROM golang:latest
```
Comment (felixge) [inline internal/apps/unit-of-work/Dockerfile:1]: @ahmed-mez Sorry for the late ping on this, but what's the security benefit of this vs specifying a known version (without sha1 to verify)? Arguably both could be problematic if somebody gains write access to the image repository?

Comment (ahmed-mez) [reply internal/apps/unit-of-work/Dockerfile:1]: It's likely for stability, the `latest` tag can include breaking changes.
I'm pretty sure @ajgajg1134 was following an internal guide to make these changes, Andrew feel free to chime in.

Comment (felixge) [reply internal/apps/unit-of-work/Dockerfile:1]: Whups, I meant to ping @ajgajg1134 🙈

**Pull request merged**